### DISCUSSION TITLE
Properly remove data from 1000s of ec volumes

### DISCUSSION THREAD

**Discussion by vrvico** [seaweedfs/seaweedfs#5745] [Q&A]
In the cluster the data of a customer (300TB) needs to be removed. It is across 1000s of EC volumes but on these volumes there is also other customers data. So I can not simply delete the volumes. What are proper steps to efficiently remove the data from these volumes? Decode entire collection, remove files via filer (Redis) or weed shell, then vacuum everything, then delete empty volumes, then ec.encode?
I think I am wrong with the steps, because timing each of the operations suggests that to remove the space taken would probably take months? Please advise.

**Comment by chrislusf** [comment]
It would be the only way if the data is not in its dedicated collection.
